core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -85,12 +93,14 @@ class AddPartitionsToTxnManager(config: KafkaConfig, client: NetworkClient, time
         topicPartitionsToError.put(new TopicPartition(topic.name, partition), error)
       }
     }
+    verificationFailureRate.mark(topicPartitionsToError.size)
     topicPartitionsToError.toMap
   }
 
   private class AddPartitionsToTxnHandler(node: Node, transactionDataAndCallbacks: TransactionDataAndCallbacks) extends RequestCompletionHandler {
     override def onComplete(response: ClientResponse): Unit = {
       // Note: Synchronization is not needed on inflightNodes since it is always accessed from this thread.
+      verificationTimeMs.update(time.milliseconds() - transactionDataAndCallbacks.earliestAdditionMs)
```

Comment:
The KIP originally included the phrasing  `This will also account for verifications that fail before the coordinator is called.`

However, on further thought, I wonder if it is better to only update the value on responses from the transaction coordinator. If not, I can also populate the metric on earlier callbacks, but it the meaning gets a little convoluted because the callback only applies to a single transaction that may not be the earliest one added for the node.

Reply:
I've decided to fix this to have an update per produce request that comes into the manager.